core/domain/prod_validation_jobs_one_off.py

```diff
@@ -1870,6 +1874,463 @@ def _get_custom_validation_functions(cls):
         return [cls._validate_node_count]
 
 
+class TopicModelValidator(BaseModelValidator):
+    """Class for validating TopicModel."""
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        return topic_services.get_topic_from_model(item)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        snapshot_model_ids = [
+            '%s-%d' % (item.id, version) for version in range(
+                1, item.version + 1)]
+        skill_ids = item.uncategorized_skill_ids
+        for subtopic in item.subtopics:
+            skill_ids = skill_ids + subtopic['skill_ids']
```

Comment:
Do you care about removing duplicates? (Can there be duplicates? I'm not sure. Is this something we should check as well?)

Reply:
That is checked in domain object validation for topic.

Reply:
But we are doing this call here for models that might potentially be invalid, right? So, maybe worth doing a list(set(...)) anyway to ensure uniqueness -- but up to you. I don't feel too strongly.

Reply:
Yes, I think you are right. It is better to avoid fetching duplicate models. I have updated this!